app/src/main/java/com/example/ohthmhyh/Habit.java

```diff
@@ -18,7 +14,7 @@
     private String description;
     private long startDate;
     private ArrayList<Days> schedule = new ArrayList<Days>();
-    private int UHID;
+    private int UHID = -1;
```

Comment:
Why is this given a default value?

Reply:
setting a default value of -1 lets me make sure that the habit has had a unique habit ID set to it when we upload it to the database. Before uploading, I check and see if the value is still -1; if it is, I change it to a unique habit id. If it is not -1, then the habit has already had its habit ID set elsewhere. basically, it allows you to not set a habit ID when you create the habit as the condition will be caught at upload time